projects/components/src/lib/directives/show-clipped-text.directive.ts

```diff
@@ -0,0 +1,278 @@
+            tip.content.style.opacity = '0';
+            tip.hideTimeout = null;
+            tip.currentDirective = null;
+        }, delay);
+    },
+
+    clearHideTimeout(): void {
+        if (tip.hideTimeout) {
+            clearTimeout(tip.hideTimeout);
+            tip.hideTimeout = null;
+        }
+    },
+
+    onTransitionEnd(e: Event): void {
+        if (tip.container.style.opacity === '0') {
+            tip.container.style.visibility = 'hidden';
+            tip.content.style.visibility = 'hidden';
+        }
+    },
+
+    /**
+     * Update the tooltip based on the content/dimensions of {@link tip.currentDirective}
+     */
+    update(): void {
+        const el = tip.currentHost;
+        const rect = el.getBoundingClientRect();
+        setStyle(tip.container, {
+            top: rect.top + 'px',
+            left: rect.left + 'px',
+            width: rect.width + 'px',
+            height: rect.height + 'px',
+            opacity: '1',
+        });
+        setStyle(tip.content, {
+            visibility: 'visible',
+            opacity: '1',
+            width: this.tooltipSize,
+        });
+        const { tl, tr, bl, br } = TooltipPosition;
+        tip.content.classList.remove(tl, tr, bl, br);
+        tip.content.classList.add(tip.calculatePosition(rect));
+        tip.content.innerHTML = el.innerHTML;
+    },
+
+    /**
+     *  Tooltip direction should be the opposite quadrant of where the center of the clipped element is
+     *     TL -> BR        TR -> BL
+     *     BL -> TR        BR -> TL
+     * @param rect THe dimensions of the clipped element
+     */
+    calculatePosition(rect: ClientRect): TooltipPosition {
+        const { innerWidth, innerHeight } = window;
+        const rectCenter = { x: rect.left + rect.width / 2, y: rect.top + rect.height / 2 };
+
+        // It's on the right
+        if (rectCenter.x > innerWidth / 2) {
+            if (rectCenter.y > innerHeight / 2) {
+                // Bottom right quad
+                return TooltipPosition.tl;
+            } else {
+                // Top right quad
+                return TooltipPosition.bl;
+            }
+        }
+
+        // It's on the left
+        if (rectCenter.y > innerHeight / 2) {
+            // Bottom left quad
+            return TooltipPosition.tr;
+        } else {
+            // Top left quad
+            return TooltipPosition.br;
+        }
+    },
+};
+
+type EventHandler = (e: MouseEvent) => void;
+
+function watchEvents(el: HTMLElement, mouseIn: EventHandler, mouseOut: EventHandler): void {
+    el.addEventListener('mouseenter', mouseIn);
+    el.addEventListener('mouseleave', mouseOut);
+}
+
+function unwatchEvents(el: HTMLElement, mouseIn: EventHandler, mouseOut: EventHandler): void {
+    el.removeEventListener('mouseenter', mouseIn);
+    el.removeEventListener('mouseleave', mouseOut);
+}
+
+function setStyle(el: HTMLElement, style: Partial<CSSStyleDeclaration>): void {
```

Comment:
Documentation about why this method is needed would be helpful

Reply:
It's clearly a small utility, private method only used here, we don't require comments for those, is something not clear?